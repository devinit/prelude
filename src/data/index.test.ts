import {toId, getTotal, normalizeKeyName, groupedValuesIntoPercents} from '.';
import * as prettyFormat from 'pretty-format';

const dataA = [
        {di_id: 'AL', value: 3000, year: 2000},
        {di_id: 'UK', value: 3000, year: 2000}
        ];

describe('Utility functions test', () => {
    it('should remove di_id field in objects and replace with id', () => {
        const formatted = toId(dataA[0]);
        expect(formatted.id).toBe('AL');
    });
    it('should get total of the value field in an array', () => {
        expect(getTotal(dataA)).toBe(6000);
    });
    it('should normalize colum name ie remove _ where necessry', () => {
        const ageBand = normalizeKeyName('value_0_14');
        expect(ageBand).toBe('0-14');
    });
    it('should turn values into percents for grouped data', () => {
        const data = [
            {sector: 'a', value: 2},
            {sector: 'b', value: 3},
            {sector: 'a', value: 4},
            {sector: 'b', value: 7}
        ];
        const result =  groupedValuesIntoPercents(data, 'sector');
        expect(prettyFormat({result})).toMatchSnapshot();
    });
});
