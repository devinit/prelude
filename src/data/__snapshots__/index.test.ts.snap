// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utility functions test should turn values into percents for grouped data 1`] = `
"Object {
  \\"result\\": Array [
    Object {
      \\"sector\\": \\"a\\",
      \\"value\\": 33,
    },
    Object {
      \\"sector\\": \\"a\\",
      \\"value\\": 67,
    },
    Object {
      \\"sector\\": \\"b\\",
      \\"value\\": 30,
    },
    Object {
      \\"sector\\": \\"b\\",
      \\"value\\": 70,
    },
  ],
}"
`;
